mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulSettingsScreen.kt

```diff
@@ -25,38 +25,46 @@ class AuthenticatedUserProfileScreenState(
     chessFacade: ChessFacade,
     scope: CoroutineScope,
     onEditProfileNameClickAction: State<() -> Unit>,
+    onEditProfileImageClickAction: State<() -> Unit>,
 ) : SettingScreenState, ProfileScreenState by StatefulProfileScreen(user, chessFacade, scope) {
   override val email = user.email
   override val puzzlesCount = 0
   private val onEditProfileNameClickAction by onEditProfileNameClickAction
+  private val onEditProfileImageClickAction by onEditProfileImageClickAction
 
-  override fun onSettingsClick() {}
   override fun onEditProfileNameClick() {
     onEditProfileNameClickAction()
   }
+  override fun onEditProfileImageClick() {
+    onEditProfileImageClickAction()
+  }
 }
 
 /**
  * A stateful composable to visit setting page of the logged-in user
  *
  * @param user the current logged-in user.
- * @param onEditProfileNameClick Callable lambda to navigate to the profile Edit popup
+ * @param onEditProfileNameClick Callable lambda to navigate to the profile name edit popup
+ * @param onEditProfileImageClick Callable lambda to navigate to the profile image edit popup
  * @param modifier the [Modifier] for this composable.
  * @param contentPadding the [PaddingValues] to apply to this screen.
  */
 @Composable
 fun StatefulSettingsScreen(
     user: AuthenticatedUser,
     onEditProfileNameClick: () -> Unit,
+    onEditProfileImageClick: () -> Unit,
     modifier: Modifier = Modifier,
     contentPadding: PaddingValues = PaddingValues(),
 ) {
   val chessFacade = LocalChessFacade.current
   val scope = rememberCoroutineScope()
   val currentOnEditProfileNameClick = rememberUpdatedState(onEditProfileNameClick)
+  val currentOnEditProfileImageClick = rememberUpdatedState(onEditProfileImageClick)
   val state =
       remember(user, chessFacade, scope, currentOnEditProfileNameClick) {
```

Comment:
Missing `currentOnEditProfileImageClick` key for `remember`.

```kotlin
      remember(user, chessFacade, scope, currentOnEditProfileNameClick, currentOnEditProfileImageClick) {
```

Reply:
thx, I missed that